### PR TITLE
Include all aliases including non-filtering in  `_search_shards` response

### PR DESCRIPTION
**Pull request by s1monw**
`_search_shards` API today only returns aliases names if there is an alias
filter associated with one of them. Now it can be useful to see which aliases
have been expanded for an index given the index expressions. This change also includes non-filtering aliases even without a filtering alias being present.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsRequest.java [hunk 1]
```diff
@@ -55,6 +56,22 @@ public ActionRequestValidationException validate() {
     }
 
     /**
+     * Iff set to <code>true</code> the response will contain all aliases expanded form the index expressions even if
+     * they don't have a filter associated with it.
+     */
+    public void setIncludeNonFilteringAliases(boolean includeNonFilteringAliases) {
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsRequest.java]: call it includeAllAliases like in the REST layer?

File: docs/reference/search/search-shards.asciidoc [hunk 2]
```diff
@@ -166,3 +166,7 @@ routing values have been specified.
     A boolean value whether to read the cluster state locally in order to
     determine where shards are allocated instead of using the Master node's
     cluster state.
```
Comment (javanna) [inline docs/reference/search/search-shards.asciidoc]: I think that we need to adjust a sentence above where we say that if filtered aliases are used, they will be returned.

**Review by javanna — APPROVED** [review]
left a small comment, LGTM otherwise.  I would propose that we treat this as a bugfix and we add all aliases in there by default, with no option to disable this behaviour. After all it makes little sense from a user perspective to return only filtered aliases and we should have had them all from day one? That would also not require any bw comp handling. I am pretty sure this change won't affect any user as it doesn't change the structure of the response. Also, the new option may sound confusing to users. My two cents.

**Comment by s1monw** [comment]
@javanna agreed - I updated the PR

File: core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java [hunk 3]
```diff
@@ -261,15 +263,28 @@ public String resolveDateMathExpression(String dateExpression) {
         return dateMathExpressionResolver.resolveExpression(dateExpression, new Context(null, null));
     }
 
+
+
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java]: remove these?

File: docs/reference/search/search-shards.asciidoc [hunk 4]
```diff
@@ -165,4 +165,4 @@ routing values have been specified.
 `local`::
     A boolean value whether to read the cluster state locally in order to
     determine where shards are allocated instead of using the Master node's
-    cluster state.
+    cluster state.
```
Comment (javanna) [inline docs/reference/search/search-shards.asciidoc]: can you update the sentence: "When filtered aliases are used, the filter is returned as part of the indices section [5.1.0] Added in 5.1.0." ?

**Review by javanna — APPROVED** [review]
left two minors, LGTM otherwise

File: docs/reference/search/search-shards.asciidoc (hunk 4 above)
Comment (s1monw) [inline docs/reference/search/search-shards.asciidoc]: done

**Pull request merged**